### ISSUE TITLE
`brew update` made destructive changes to my system without asking

### ISSUE THREAD

**Issue by josh-berry** [Homebrew/homebrew-core#9316]
When I run `brew update`, I expect Homebrew to update its own package list and perform any internal housekeeping tasks related to this. I do **not** expect it to install or uninstall software without my consent.

However, the last time I ran `brew update`, this happened:

```
$ brew update
Updated Homebrew from d7c463ad2 to 78d584f42.
Updated 4 taps (caskroom/cask, homebrew/core, homebrew/gui, homebrew/versions).
==> New Formulae
[snip]
==> Renamed Formulae
[snip]
==> Deleted Formulae
dynamodb-local                           homebrew/gui/inkscape ✔
[snip]
==> inkscape has been moved to Homebrew-Cask.
==> brew uninstall --force inkscape
Uninstalling inkscape... (1,981 files, 221.7M)
==> brew prune
Pruned 18 symbolic links and 5 directories from /usr/local
==> brew cask install inkscape
==> Satisfying dependencies
Error: inkscape requires XQuartz/X11, which can be installed using Homebrew-Cask by running
  brew cask install xquartz

or manually, by downloading the package from
  https://www.xquartz.org/
Error: nothing to install
```

As you can see above, Inkscape is now no longer installed on my system, yet I did not ask for it to be removed.  (Obviously this is a bug, since it should have been reinstalled, but this issue is not about that bug.)

Please ask for user consent before making potentially destructive changes such as attempting to uninstall and reinstall a package.

**Comment by ilovezfs** [comment]
See
https://github.com/Homebrew/homebrew-gui/issues/53
and
https://github.com/Homebrew/homebrew-gui/commit/c05704f461d8b806430b140ef98fbe50420f0dfb

**Comment by ilovezfs** [comment]
That is called a "migration" and may happen at any time during `brew update`. TIL

**Comment by josh-berry** [comment]
Whatever it's called, it shouldn't have been done without asking (or an explicit instruction from the user e.g. `brew upgrade`, `brew install`, etc.).

**Comment by MikeMcQuaid** [comment]
@josh-berry We disagree. `brew update` also performs migrations and this isn't going to change. Sorry that you were inconvenienced.

**Comment by josh-berry** [comment]
@MikeMcQuaid In that case, please provide users adequate notice that this is expected behavior of `brew update`. Nowhere in the manpage or any documentation that I can find does it mention that `brew update` might make changes to your installed system configuration.

The manpage says:

```
       update [--merge] [--force]
              Fetch the newest version of Homebrew  and  all  formulae  from  GitHub
              using git(1).

              If  --merge  is  specified  then  git merge is used to include updates
              (rather than git rebase).

              If --force is specified then always do a  slower,  full  update  check
              even if unnecessary.
```

The homepage (brew.sh) does not even mention `brew update`. The FAQ in the `docs/` subdirectory mentions it in passing but does not explain what migrations are.

In short, there was no way--short of reading the source code--for me to even discover that this is intended behavior, leading to a rather unpleasant surprise as described above.

**Comment by MikeMcQuaid** [comment]
@josh-berry A reminder that you're using software provided to you for free and ["`PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED`"](https://github.com/Homebrew/brew/blob/master/LICENSE.txt).

`brew update` has been performing various migrations for years at this point. I'm sorry that you've been inconvenienced by this one and that our documentation is lacking on this point but we're a understaffed and volunteer run project so bugs and incomplete documentation are inevitable.
